core/src/main/java/net/adoptopenjdk/icedteaweb/client/util/UiLock.java

```diff
@@ -0,0 +1,43 @@
+package net.adoptopenjdk.icedteaweb.client.util;
+
+import net.adoptopenjdk.icedteaweb.Assert;
+import net.sourceforge.jnlp.config.DeploymentConfiguration;
+
+import javax.swing.JComboBox;
+import javax.swing.text.JTextComponent;
+import java.awt.Component;
+import java.util.Optional;
+
+/**
+ * Helper class that can be used to check if a UI element should be locked / disabled in the UI
+ */
+public class UiLock {
+
+    private final DeploymentConfiguration deploymentConfiguration;
+
+    private final static String LOCK_SUFFIX = ".locked";
+
+    public UiLock(final DeploymentConfiguration deploymentConfiguration) {
+        this.deploymentConfiguration = Assert.requireNonNull(deploymentConfiguration, "deploymentConfiguration");
+    }
+
+    public void update(final String propertyName, final Component component) {
+        final boolean isLocked = Optional.ofNullable(deploymentConfiguration.getProperty(propertyName + LOCK_SUFFIX))
```

Comment:
Would be better to add a method `isLocked(String)` to the DeploymentConfiguration and make use of the `currentConfiguration.get(key).isLocked()` which is available inside of DeploymentConfiguration

Reply:
What do you mean? `DeploymentConfiguration` has a `getProperty(key)` method that returns a string. There is not `get(key)` method.